dataland-e2etests/src/test/kotlin/org/dataland/e2etests/TestDataProvider.kt

```diff
@@ -0,0 +1,83 @@
+package org.dataland.e2etests
+
+import com.squareup.moshi.FromJson
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.ToJson
+import com.squareup.moshi.Types
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import org.dataland.datalandbackend.openApiClient.model.CompanyAssociatedDataEuTaxonomyData
+import org.dataland.datalandbackend.openApiClient.model.CompanyInformation
+import org.dataland.datalandbackend.openApiClient.model.EuTaxonomyData
+import org.springframework.core.io.ClassPathResource
+import java.lang.reflect.ParameterizedType
+import java.math.BigDecimal
+import java.time.LocalDate
+
+object BigDecimalAdapter {
+    @FromJson
+    fun fromJson(string: String) = BigDecimal(string)
+
+    @ToJson
+    fun toJson(value: BigDecimal) = value.toString()
+}
+
+object LocalDateAdapter {
+    @FromJson
+    fun fromJson(string: String) = LocalDate.parse(string)
+
+    @ToJson
+    fun toJson(value: LocalDate) = value.toString()
+}
```

Comment:
Kann mir das bitte jmd erklären, inklusive Annotations?
Habe ich wohl verpasst und würde es gerne verstehen.

Reply:
Das sind die beiden adapter für localdate und bigdecimal welche von moshi leider nicht nativ unterstützt werden. Wir brauchen ja weiter unten im Skript einen adapter der json in model und model in json parst. Hier wird dem adapter jetzt gesagt was er tun soll wenn model in json formatiert wird (@ToJson) und json in model (@FromJson). Das geht bei der fasterxml library die wir im backend haben out of the box. Der openapi generator verwendet aber moshi und da geht es leider nicht out of the box deswegen hier die custom adapter.

Reply:
Alles klar, Danke für die Erklärung 👍